.github/workflows/core.yml

```diff
@@ -0,0 +1,549 @@
+      - run: chmod +x core/build/unix/trezor-emu-core*
+      - uses: ./.github/actions/environment
+      - run: nix-shell --run "tests/download_emulators.sh"
+      - run: nix-shell --run "poetry run pytest tests/upgrade_tests"
+
+
+  # Persistence tests - UI.
+  core_persitence_test:
+    name: Persistence tests
+    runs-on: ubuntu-latest
+    needs: core_emu
+    strategy:
+      matrix:
+        model: [T2T1]  # TODO T2B1 https://github.com/trezor/trezor-firmware/issues/2724
+        asan: ${{ fromJSON(github.event_name == 'schedule' && '["noasan", "asan"]' || '["noasan"]') }}
+    env:
+      TREZOR_PROFILING: ${{ matrix.asan == 'noasan' && '1' || '0' }}
+      PYTEST_TIMEOUT: 400
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions/download-artifact@v3
+        with:
+          name: core-emu-${{ matrix.model }}-universal-debuglink-${{ matrix.asan }}
+          path: core/build
+      - run: chmod +x core/build/unix/trezor-emu-core*
+      - uses: ./.github/actions/environment
+      - run: nix-shell --run "poetry run make -C core test_emu_persistence_ui"
+        if: ${{ matrix.asan == 'noasan' }}
+      - run: nix-shell --run "poetry run make -C core test_emu_persistence"
+        if: ${{ matrix.asan == 'asan' }}
+      - uses: ./.github/actions/ui-report
+        with:
+          artifact-name: core-test-persistence-${{ matrix.model }}-${{ matrix.asan }}
+      - run: mv core/src/.coverage core/.coverage.test_persistence || true
+      - uses: actions/upload-artifact@v3
+        with:
+          name: core-coverage-${{ matrix.model }}
+          path: core/.coverage.*
+          retention-days: 7
+
+  core_hwi_test:
+    name: HWI tests
+    if: false  # XXX currently failing
+    continue-on-error: true
+    runs-on: ubuntu-latest
+    needs: core_emu
+    strategy:
+      matrix:
+        model: [T2T1]  # TODO T2B1
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions/download-artifact@v3
+        with:
+          name: core-emu-${{ matrix.model }}-universal-debuglink-noasan
+          path: core/build
+      - run: chmod +x core/build/unix/trezor-emu-core*
+      - uses: ./.github/actions/environment  # XXX poetry maybe not needed
+      - run: nix-shell --run "git clone --depth=1 https://github.com/bitcoin-core/HWI.git"
+      # see python_test for explanation of _PYTHON_SYSCONFIGDATA_NAME
+      - run: nix-shell --arg fullDeps true --run "unset _PYTHON_SYSCONFIGDATA_NAME && cd HWI && poetry install && poetry run ./test/test_trezor.py --model_t ../core/build/unix/trezor-emu-core bitcoind"
+      - uses: actions/upload-artifact@v3
+        with:
+          name: core-test-hwi-${{ matrix.model }}
+          path: HWI/trezor-t-emulator.stdout
+          retention-days: 7
+
+  core_memory_profile:
+    name: Memory allocation report
+    if: false  # NOTE manual job, comment out to run
```

Comment:
how would this work, you would uncomment it every time you want it to run or is this just for testing purpose for now? I think we cant have manual jobs in one workflow, it would need to be a separate one.

Reply:
Yes, needs to be uncommented before every run, as it's very rarely used. I may turn it into a manual workflow later.